### PR TITLE
Sak 42229: Drag and drop implemented

### PR DESCRIPTION
**Pull request by ajordanedf**

### CODE REVIEW

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm [hunk 1]
```diff
@@ -573,4 +573,4 @@ if ( $(window).width() > portalSmallBreakPoint() ) { ## Don't go modal when mobi
 		$("#portletBody").dialog("option", "width", modalDialogWidth());
 	});
 }
-</script>
+</script>
```
Comment (mpellicer) [inline content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: Clean this change.

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm [hunk 2]
```diff
@@ -138,19 +138,17 @@
...
-						</div>
+					<div class="form-inline" style="text-align: center;">
+						<label style="width:100%; border: 2px dashed #424242 !important; border-radius: 5px; min-height: 150px;
+						 display: flex; justify-content: center; flex-direction: column; align-items: center; position: relative">
+							<input type="file" style="opacity: 0; position: absolute ;width:100%; height:100%; z-index: 1" name="upload" id="upload" class="upload" 
+							onchange="showNotif('submitnotifxxx','attachButton','attachForm'); $('#submitnotifxxx').removeClass('hidden'); document.getElementById('attachForm').action='#toolLinkParam("FilePickerAction" "doAttachupload" "special=upload&sakai_csrf_token=$validator.escapeUrl($sakai_csrf_token)")';submitform('attachForm');" #if($disable_attach_links) disabled="disabled"#end />
+							<span class="btn btn-default" type="button" style="position: absolute;">$tlang.getString("span.dragButton")</button>
+						</label>
+						<span id="submitnotifxxx" class="messageProgress hidden" style="margin-bottom: 0; visibility: hidden;">$tlang.getString("processmessage.file")</span>
```
Comment (fostersdesign) [inline content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: Can these inline styles be extracted to CSS rules and classes, please?

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm [hunk 3]
```diff
@@ -138,19 +138,17 @@
...
-					<div class="form-inline">
-						<div class="form-group">
-						    <label for="upload">$tlang.getString("att.upl")</label>
-						    <input type="file" name="upload" id="upload" class="upload" onchange="showNotif('submitnotifxxx','attachButton','attachForm'); document.getElementById('attachForm').action='#toolLinkParam("FilePickerAction" "doAttachupload" "special=upload&sakai_csrf_token=$validator.escapeUrl($sakai_csrf_token)")';submitform('attachForm');" #if($disable_attach_links) disabled="disabled"#end />
-							<span id="submitnotifxxx" class="messageProgress" style="visibility:hidden">$tlang.getString("processmessage.file")</span>
-						</div>
-						<div class="form-group">
-							<span id="submitnotifxxx" class="messageProgress" style="visibility:hidden">$tlang.getString("processmessage.file")</span>
-						</div>
+					<div class="form-inline" style="text-align: center;">
```
Comment (fostersdesign) [inline content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: Please extract these inline styles to classes.

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm (hunk 2 above)
Comment (fostersdesign) [inline content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: Please use the .sak-banner-info class instead of .messageProgress.

**Pull request closed without merging**